core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,285 @@
+     * for receiving {@link PollResponse} in async manner.
+     */
+    public Mono<PollResponse<T>> poll() {
+        return this.pollOperation.apply(this.pollResponse)
+            .doOnEach(pollResponseSignal -> {
+                if (pollResponseSignal.get() != null) {
+                    this.pollResponse = pollResponseSignal.get();
+                }
+            });
+    }
+
+    /**
+     * Blocks execution and wait for polling to complete. The polling is considered finished based on status defined in {@link com.azure.core.util.polling.PollResponse.OperationStatus}.
+     * Auto polling must be turned on for poller to continuously poll in background.
+     *
+     * @return returns final poll response when polling is finished as defined in {@link com.azure.core.util.polling.PollResponse.OperationStatus}.
+     */
+    public PollResponse<T> block() {
+        return this.fluxHandle.blockLast();
+    }
+
+    /*
+     * This function will apply delay and call poll operation function async.
+     * We expect Mono from pollOperation should never return Mono.error() . If any unexpected
+     * scenario happens in pollOperation, it should catch it and return a valid PollResponse.
+     * This is because poller does not know what to do in case on Mono.error.
+     * This function will return empty mono in case of Mono.error() returned by poll operation.
+     *
+     * @return mono of poll response
+     */
+    private Mono<PollResponse<T>> asyncPollRequestWithDelay() {
+        return  Mono.defer(() -> this.pollOperation.apply(this.pollResponse)
+            .delaySubscription(getCurrentDelay())
+            .onErrorResume(throwable -> {
+                // We should never get here and since we want to continue polling
+                //Log the error
+                return Mono.empty();
+            })
+            .doOnEach(pollResponseSignal -> {
+                if (pollResponseSignal.get() != null) {
+                    this.pollResponse = pollResponseSignal.get();
+                }
+            }));
+    }
+
+    private Duration getCurrentDelay() {
+        return (this.pollResponse != null && this.pollResponse.getRetryAfter() != null) ? this.pollResponse.getRetryAfter() : this.pollInterval;
+    }
+
+    /**
+     * Provide control to turn auto polling <strong>on or off</strong>. Once auto polling is turned off, it is <strong>user's responsibility</strong>
+     * to turn it back on.
+     *
+     * <p><strong>Disable auto polling</strong></p>
+     * {@codesnippet com.azure.core.util.polling.poller.disableautopolling}
+     *
+     * <p><strong>Enable auto polling</strong></p>
+     * {@codesnippet com.azure.core.util.polling.poller.enableautopolling}
+     *
+     * @param autoPollingEnabled true  Ensures the polling is happening in background.
+     * false Ensures that polling is <strong>not</strong> happening in background.
```

Comment:
@param autoPollingEnabled If true, auto-polling will occur transparently in the background, otherwise it requires manual polling by the user to get the latest state.

Reply:
resolved